### ISSUE TITLE
SheetAccessedUsingStringInspection inspection can be made generic

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#4470]
The `SheetAccessedUsingStringInspection` is currently Excel-specific. However in my tests for other things, I observed that all document modules from various Office hosts were always predeclared. I believe by definition, a document module must be predeclared. In this case, it's then safe to expect that it be accessed using identifiers rather than names. 

There might be problems with using default instances (on the same level as doing `UserForm1.Show`), but I think that if they are going to use a default instance, they should at least use an identifier. 

So, I would propose that the inspection be revised to inspect the object model for any collections that returns a type that is a document module, then flag those using strings rather than retrieving the object directly.

**Comment by comintern** [comment]
I agree that this inspection could be expanded, but I think the main problem with determining with collection members to inspect is that they are more likely to take a `Variant` as an argument rather than a `String`.  For example, `IWorksheets.Item`:
```
        [propget, helpcontext(0x00072c3e)]
        HRESULT _stdcall Item(
                        [in] VARIANT Index, 
                        [out, retval] IDispatch** RHS);
```
This is likely to support code such as:
```
Sub Foo()
    Dim ws As Worksheet
    Set ws = Worksheets(Sheet1.Cells(1, 1))
    Debug.Print ws.Name
End Sub
```
I suspect that most of the other hosts\libraries would have similar signatures, so it may be pretty difficult to identify the members that would be inspection targets.  That said, I have no problem with using hard-coded criteria for specific hosts.

**Comment by MDoerner** [comment]
I think this is a `Variant` parameter because you can pass in the ordinal index as well:
`Set ws = WorkSheets(1)` sets  `ws` to the first worksheet in the active workbook.
